src/repository/storage/postgres/orm/sequelize/noteRelations.ts

```diff
@@ -209,4 +210,43 @@ export default class NoteRelationsSequelizeStorage {
 
     return foundNote !== null;
   };
+
+  /**
+   * Get all parent notes of a note that a user has access to,
+   * where the user has access to.
+   * @param noteId - the ID of the note.
+   */
+  public async getNoteParentsIds(noteId: NoteInternalId): Promise<NoteInternalId[]> {
+    let parentNotes: NoteInternalId[] = [];
+
+    // get all note ids via a singe sql query instead of many
+    const query = `
+    WITH RECURSIVE note_parents AS (
+      SELECT np.note_id, np.parent_id
+      FROM ${String(this.database.literal(this.tableName).val)} np
+      WHERE np.note_id = :startNoteId
+      UNION ALL
+      SELECT nr.note_id, nr.parent_id
+      FROM ${String(this.database.literal(this.tableName).val)} nr
+      INNER JOIN note_parents np ON np.parent_id = nr.note_id
+    )
+    SELECT np.note_id, np.parent_id
+    FROM note_parents np;`;
+
+    try {
+      const result = await this.database.query(query, {
+        replacements: { startNoteId: noteId },
+        type: QueryTypes.SELECT,
+      });
+
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      parentNotes = (result as { note_id: number; parent_id: number }[])?.map(note => note.parent_id) ?? [];
```

Comment:
why `parent_id` and not `note_id`? What is "parentNotes"?

Reply:
we need `parent_id` because we don't want to include the main `noteId` in the result. The `parentNotes` variable is where we save all the note parents ids that would be returned. I will rename it to `noteParents`